src/data/mystery-encounters/encounters/fiery-fallout-encounter.ts

```diff
@@ -233,26 +243,27 @@ export const FieryFalloutEncounter: MysteryEncounter =
             { fillRemaining: true },
             undefined,
             () => {
-              giveLeadPokemonCharcoal(scene);
+              giveLeadPokemonAttackTypeBoostItem(scene);
             });
 
           const primary = encounter.options[2].primaryPokemon!;
-          const secondary = encounter.options[2].secondaryPokemon![0];
 
-          setEncounterExp(scene, [ primary.id, secondary.id ], getPokemonSpecies(Species.VOLCARONA).baseExp * 2);
+          setEncounterExp(scene, [ primary.id ], getPokemonSpecies(Species.VOLCARONA).baseExp * 2);
           leaveEncounterWithoutBattle(scene);
         })
         .build()
     )
     .build();
 
-function giveLeadPokemonCharcoal(scene: BattleScene) {
-  // Give first party pokemon Charcoal for free at end of battle
+function giveLeadPokemonAttackTypeBoostItem(scene: BattleScene) {
+  // Give first party pokemon attack type boost item for free at end of battle
   const leadPokemon = scene.getParty()?.[0];
   if (leadPokemon) {
-    const charcoal = generateModifierType(scene, modifierTypes.ATTACK_TYPE_BOOSTER, [ Type.FIRE ]) as AttackTypeBoosterModifierType;
-    applyModifierTypeToPlayerPokemon(scene, leadPokemon, charcoal);
-    scene.currentBattle.mysteryEncounter!.setDialogueToken("leadPokemon", leadPokemon.getNameToRender());
-    queueEncounterMessage(scene, `${namespace}:found_charcoal`);
+    const encounter = scene.currentBattle.mysteryEncounter!;
+    const boosterModifierType = generateModifierType(scene, modifierTypes.ATTACK_TYPE_BOOSTER) as AttackTypeBoosterModifierType;
```

Comment:
```suggestion
    // Generate type booster held item, default to Charcoal if item fails to generate
    const boosterModifierType = (generateModifierType(scene, modifierTypes.ATTACK_TYPE_BOOSTER)
      ?? generateModifierType(scene, modifierTypes.ATTACK_TYPE_BOOSTER, [ Type.FIRE ])) as AttackTypeBoosterModifierType;
```
Failsafe in case the type booster held item somehow fails to generate (e.g., if the party's Pokemon have only status moves)